tsdb/engine/tsm1/compact_test.go

```diff
@@ -4180,78 +4032,272 @@ func TestIsGroupOptimized(t *testing.T) {
 }
 
 func TestEnginePlanCompactions(t *testing.T) {
-	testFileSets := [][]tsm1.FileStat{
-		[]tsm1.FileStat{
-			{
-				Path: "01-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "02-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "03-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "04-04.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-		},
-		[]tsm1.FileStat{
-			{
-				Path: "01-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "02-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "03-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-			{
-				Path: "04-05.tsm",
-				Size: 256 * 1024 * 1024,
-			},
-		},
-	}
-	testBlockCountsAndResults := []struct {
-		blockCounts  []int
+	type testLevelResults struct {
 		level1Groups []tsm1.PlannedCompactionGroup
 		level2Groups []tsm1.PlannedCompactionGroup
 		level3Groups []tsm1.PlannedCompactionGroup
 		level4Groups []tsm1.PlannedCompactionGroup
 		level5Groups []tsm1.PlannedCompactionGroup
-	}{
+	}
+
+	type testEnginePlanCompactionsRunner struct {
+		name        string
+		files       []tsm1.FileStat
+		blockCounts []int
+		// This is specifically used to adjust the modification time
+		// so we can simulate the passage of time in tests
+		testShardTime time.Duration
+		// Each result is for the different plantypes
+		getResultByPlanType func(planType tsm1.PlanType) testLevelResults
+	}
+
+	tests := []testEnginePlanCompactionsRunner{
 		{
+			name: "many generations under 2GB",
+			files: []tsm1.FileStat{
+				{
+					Path: "01-05.tsm",
+					Size: 256 * 1024 * 1024,
+				},
+				{
+					Path: "02-05.tsm",
+					Size: 256 * 1024 * 1024,
+				},
+				{
+					Path: "03-05.tsm",
+					Size: 256 * 1024 * 1024,
+				},
+				{
+					Path: "04-04.tsm",
+					Size: 256 * 1024 * 1024,
+				},
+			},
 			blockCounts: []int{
 				tsdb.DefaultAggressiveMaxPointsPerBlock,
 				tsdb.DefaultMaxPointsPerBlock,
 				tsdb.DefaultMaxPointsPerBlock,
 				tsdb.DefaultMaxPointsPerBlock,
 			},
-			level5Groups: []tsm1.PlannedCompactionGroup{
-				{
-					tsm1.CompactionGroup{"01-05.tsm", "02-05.tsm", "03-05.tsm", "04-04.tsm"},
-					tsdb.DefaultAggressiveMaxPointsPerBlock,
-				},
+			testShardTime: -1,
+			getResultByPlanType: func(planType tsm1.PlanType) testLevelResults {
+				switch planType {
+				case tsm1.PT_Standard:
+					return testLevelResults{
+						level4Groups: []tsm1.PlannedCompactionGroup{
+							{
+								tsm1.CompactionGroup{"01-05.tsm", "02-05.tsm", "03-05.tsm", "04-04.tsm"},
+								tsdb.DefaultMaxPointsPerBlock,
+							},
+						},
+					}
+				case tsm1.PT_SmartOptimize:
```

Comment:
For ease of reading, I would make the case 

`case tsm1.PT_SmartOptimize, tsm1.PT_NoOptimize:`

if you have two identical results.  Later changes which differentiate the cases can be made by dividing the case branches as necessary.